src/index.js

```diff
@@ -63,23 +74,11 @@ class MediaQuery extends React.Component {
 
     this._mql = matchMedia(this.query, values)
     this._mql.addListener(this.updateMatches)
-    this.updateMatches()
   }
 
-  componentWillUpdate(_, nextState) {
```

Comment:
This breaks the behavior of onBeforeChange - it is supposed to be executed before the render happens (hence the name).

Reply:
https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#side-effects-on-props-change
As I understand componentDidUpdate new place for that.